modules/contact/src/ContactAction.C

```diff
@@ -90,7 +90,8 @@ ContactAction::act()
       InputParameters params = _factory.getValidParams("MechanicalContactConstraint");
 
       // Extract global params
-      _app.parser().extractParams(_name, params);
+      if (isParamValid("parser_syntax"))
+        _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
If "parser_syntax" doesn't exist then there is nothing to extract. In this case, the user created the action by calling ActionFactory::create directly.

Throughout this PR I have separated the "name" (which is what is passed to *Factory::create(...)) and "parser_syntax". If the parser is creating the object, then "parser_syntax" is added to the parameters. The name is now arbitrary and has no connection to "parser_syntax" per se.